### PR TITLE
Add class and Materiel services 

### PR DESCRIPTION
**Pull request by jobrane33**
this needs a serious review please be critical 
https://eduvisionesprit.atlassian.net/browse/ES-8

### CODE REVIEW

File: src/tn/eduVision/services/MaterielService.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+ *
+ * @author job_j
+ */
+public class MaterielService implements Iservices<Materiel>{
+
+     private final Connection _connection = SqlConnectionManager.getConnection();
+     private static final Logger _logger = CustomLogger.getInstance().getLogger();
+     private static PreparedStatement statement = null;
+    
+    @Override
```
Comment (SeifEddineJridi) [inline src/tn/eduVision/services/MaterielService.java:31]: why override ?

File: src/tn/eduVision/services/SallesService.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+import tn.eduVision.entités.TypeSalle;
+import tn.eduVision.exceptions.NoDataFoundException;
+import tn.eduVision.tools.CustomLogger;
+import tn.eduVision.tools.SqlConnectionManager;
+
+/**
+ *
+ * @author job_j
+ */
+public class SallesService implements Iservices<Salle> {
```
Comment (SeifEddineJridi) [inline src/tn/eduVision/services/SallesService.java:26]: also IService is better than Iservices , naming not a big deal . also for betting califying should we change methods to start with capital letters ? Add , Update....

File: src/tn/eduVision/services/SallesService.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        catch(SQLException ex){
+            _logger.log(Level.SEVERE, ex.getMessage());
+        }
+        finally{
+            CloseStatment(statement);
+        }
+    }
+
+    @Override
+    public void delete(Salle salle) {
```
Comment (SeifEddineJridi) [inline src/tn/eduVision/services/SallesService.java:83]: this is actually what i wanted to discusss , some entities have Lists that are fetched from the database , if we go with this approach (elli chofneh deja fil classe) it needs a workaround (like initilaizing the lists as empty lists) or we can make a dto for creating and updating and for deleting we just use the id. What do you think ?

**Comment by SeifEddineJridi** [comment]
Also e5er 7aja wlh 😀 the .class files are still there and updating

File: src/tn/eduVision/services/MaterielService.java [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+                    + "WHERE `ressources`.`id_ressource` = ?;";
+            statement = _connection.prepareStatement(updateMateriel);
+            statement.setString(1, materiel.getNomMateriel());
+            statement.setInt(2, materiel.getQuantite());
+            statement.setInt(3, materiel.getIdRessource());
+            
+            int rowsAffected = statement.executeUpdate();
+            
+            if(rowsAffected == 0){
+                _logger.log(Level.WARNING, "No data been updated !");
```
Comment (SeifEddineJridi) [inline src/tn/eduVision/services/MaterielService.java:73]: "Materielle Non Trouvee" or something like lets not mix 2 languages  and lets be specific What do you think ?

File: src/tn/eduVision/services/MaterielService.java (hunk 1 above)
Comment (jobrane33) [reply src/tn/eduVision/services/MaterielService.java:31]: @SeifEddineJridi when i auto impliment from the interface tji b @override so mich ena zedtha won't be a problem

File: src/tn/eduVision/services/SallesService.java (hunk 2 above)
Comment (jobrane33) [reply src/tn/eduVision/services/SallesService.java:26]: i'll update it fi master

File: src/tn/eduVision/services/SallesService.java (hunk 3 above)
Comment (jobrane33) [reply src/tn/eduVision/services/SallesService.java:83]: make a PR warini example also if you can

Comment (jobrane33) [reply src/tn/eduVision/services/SallesService.java:83]: also i think we should always keep method te3hom and keep it simple ama as u wish

File: src/tn/eduVision/services/MaterielService.java (hunk 4 above)
Comment (jobrane33) [reply src/tn/eduVision/services/MaterielService.java:73]: taw nbadel bl fr ama logger y3tini name class eli loggit mnha so bch n3rfha Materielle 😄

**Review by SeifEddineJridi — APPROVED** [review]

**Comment by SanaBenHammouda** [comment]
.

**Review by ImenSebteoui — APPROVED** [review]

**Comment by Majdi-Zlitni** [comment]
> this needs a serious review please be critical https://eduvisionesprit.atlassian.net/browse/ES-8

Are all member required to review?

**Review by Majdi-Zlitni — APPROVED** [review]
Well done

**Pull request merged**